src/_P121_HMC5883L.ino

```diff
@@ -0,0 +1,158 @@
+
+/*******************************************************************************
+* Copyright 2021
+* Written by Sven Briesen (adsorstuff@gmail.com)
+* BSD license, all text above must be included in any redistribution
+*
+* Release notes:
+* Adafruit_HMC5883_Unified Library v1.2.0 required (https://github.com/adafruit/Adafruit_HMC5883_Unified)
+* Used P106 BME680 as starting point
+/******************************************************************************/
+
+#include "src/PluginStructs/P121_data_struct.h"
+#include <Adafruit_Sensor.h>
+#include <Adafruit_HMC5883_U.h>
+
+#define PLUGIN_121
+#define PLUGIN_ID_121 121
+#define PLUGIN_NAME_121 "Environment - HCM5883L"
+#define PLUGIN_VALUENAME1_121 "x"
+#define PLUGIN_VALUENAME2_121 "y"
+#define PLUGIN_VALUENAME3_121 "z"
+#define PLUGIN_VALUENAME4_121 "dir"
+
+boolean Plugin_121(uint8_t function, struct EventStruct *event, String &string)
+{
+  boolean success = false;
+
+  switch (function)
+  {
+  case PLUGIN_DEVICE_ADD:
+  {
+    Device[++deviceCount].Number = PLUGIN_ID_121;
+    Device[deviceCount].Type = DEVICE_TYPE_I2C;
+    Device[deviceCount].VType = Sensor_VType::SENSOR_TYPE_QUAD;
+    Device[deviceCount].Ports = 0;
+    Device[deviceCount].PullUpOption = false;
+    Device[deviceCount].InverseLogicOption = false;
+    Device[deviceCount].FormulaOption = true;
+    Device[deviceCount].ValueCount = 4;
+    Device[deviceCount].SendDataOption = true;
+    Device[deviceCount].TimerOption = true;
+    Device[deviceCount].GlobalSyncOption = true;
+    success = true;
+    break;
+  }
+
+  case PLUGIN_GET_DEVICENAME:
+  {
+    string = F(PLUGIN_NAME_121);
+    success = true;
+    break;
+  }
+
+  case PLUGIN_GET_DEVICEVALUENAMES:
+  {
+    strcpy_P(ExtraTaskSettings.TaskDeviceValueNames[0], PSTR(PLUGIN_VALUENAME1_121));
+    strcpy_P(ExtraTaskSettings.TaskDeviceValueNames[1], PSTR(PLUGIN_VALUENAME2_121));
+    strcpy_P(ExtraTaskSettings.TaskDeviceValueNames[2], PSTR(PLUGIN_VALUENAME3_121));
+    strcpy_P(ExtraTaskSettings.TaskDeviceValueNames[3], PSTR(PLUGIN_VALUENAME4_121));
+    success = true;
+    break;
+  }
+  case PLUGIN_WEBFORM_SHOW_I2C_PARAMS:
+  {
+    success = true;
+    break;
+  }
+  case PLUGIN_I2C_HAS_ADDRESS:
+  {
+    success = (event->Par1 == 0x1E);
+    break;
+  }
+  case PLUGIN_WEBFORM_LOAD:
+  {
+    addFormFloatNumberBox(F("DeclinationAngle"), F("plugin_121_HMC5883L_decl"), PCONFIG(0), -2.0f,2.0f ,5, 0.00001f);
+    addUnit(F("degree"));
+    success = true;
+    break;
+  }
+
+  case PLUGIN_WEBFORM_SAVE:
+  {
+    PCONFIG(0) = getFormItemFloat(F("plugin_121_HMC5883L_decl"));
+    success = true;
+    break;
+  }
+
+  case PLUGIN_INIT:
+  {
+    initPluginTaskData(event->TaskIndex, new (std::nothrow) P121_data_struct());
+    P121_data_struct *P121_data =
+        static_cast<P121_data_struct *>(getPluginTaskData(event->TaskIndex));
+
+    if (nullptr != P121_data)
+    {
+      P121_data->initialized = false; // Force re-init just in case the address changed.
```

Comment:
I can't see why this is needed, as you just recreated the object, the value should be set to false as that's how you implemented it in its .h file.
So just curious to why this is needed to set to false again.